client/core/account.go

```diff
@@ -208,24 +208,58 @@ func (c *Core) AccountImport(pw []byte, acct *Account, bonds []*db.Bond) error {
 
 // UpdateCert attempts to connect to a server using a new TLS certificate. If
 // the connection is successful, then the cert in the database is updated.
+// However, it will attempt to reconnect to dex using the old cert if the new
+// connection fails.
 func (c *Core) UpdateCert(host string, cert []byte) error {
-	accountInfo, err := c.db.Account(host)
+	acct, err := c.db.Account(host)
 	if err != nil {
 		return err
 	}
 
-	accountInfo.Cert = cert
+	var connected bool
+	defer func(cert []byte) {
+		if !connected {
+			// Attempt to reconnect using old cert.
+			acct.Cert = cert
+			c.connectAccount(acct)
+		}
+	}(acct.Cert)
+
+	// Disconnect previous dex connection first but leave it in the map so it
+	// remains listed incase we need it in the interim.
+	c.connMtx.RLock()
+	if dc := c.conns[host]; dc != nil {
+		dc.connMaster.Disconnect()
+		dc.acct.lock()
+		dc.booksMtx.Lock()
```

Comment:
Looking around, I notice that this would be the only place with the nested `c.connMtx.RLock -> dc.booksMtx.Lock`. As a general rule, I like to avoid nested mutex locking if there's not a compelling reason. Any reason we can't do 👇 ?

```Go
c.connMtx.RLock()
dc := c.conns[host]
c.connMtx.RUnlock()
if dc != nil {
```

Reply:
Yes we can do that. No compelling reason except to save a LOC.

Reply:
Well I think we can do `dc, _ := c.dex(host)` as well.